### PR TITLE
[Otter] Update interface with Otter to reflect upstream changes

### PR DESCRIPTION
**Pull request by adamtuft**
This PR is for review by @LonelyCat124 

This PR updates the interface to Otter to reflect the latest changes to Otter in [v0.2.0](https://github.com/Otter-Taskification/otter/tree/v0.2.0). These changes include:

- Renaming several Otter API entrypoints,
- Changing the arguments passed to several entrypoints,
- Removing the `otterTaskSingle[Begin|End]` entrypoint which no longer exists.

### CODE REVIEW

**Comment by LonelyCat124** [comment]
@arporter @sergisiso Is there a way to override the testing system to work on PRs that don't involve master?

**Comment by arporter** [comment]
You can add "[skip ci]" to the commit message if that's what you're after?

**Comment by LonelyCat124** [comment]
The opposite, I want ci to run on this branch before I merge it into my branch. I can do it manually if needed but was hoping it could be done here.

File: src/psyclone/tests/psyir/nodes/otter_nodes_test.py [hunk 1]
```diff
@@ -186,10 +157,7 @@ def test_otterloopnode_lower_to_language():
     calls = routine.walk(Call)
     assert len(calls) == 2
     assert calls[0].routine.name == "fortran_otterLoopBegin_i"
-    assert len(calls[0].children) == 3
-    assert calls[0].children[0].name == "__FILE__"
-    assert calls[0].children[1].value == "my_routine"
-    assert calls[0].children[2].name == "__LINE__"
+    assert len(calls[0].children) == 0
```
Comment (LonelyCat124) [inline src/psyclone/tests/psyir/nodes/otter_nodes_test.py:160]: Hi Adam,

Not sure whats happened here but this test is broken - looks like you changed some stuff here to do with the calls for the argument, might wanted to revert these lines.

File: src/psyclone/tests/psyir/nodes/otter_nodes_test.py [hunk 2]
```diff
@@ -215,10 +183,7 @@ def test_otterloopiterationnode_lower_to_language():
     calls = routine.walk(Call)
     assert len(calls) == 2
     assert calls[0].routine.name == "fortran_otterLoopIterationBegin_i"
-    assert len(calls[0].children) == 3
-    assert calls[0].children[0].name == "__FILE__"
-    assert calls[0].children[1].value == "my_routine"
-    assert calls[0].children[2].name == "__LINE__"
+    assert len(calls[0].children) == 0
```
Comment (LonelyCat124) [inline src/psyclone/tests/psyir/nodes/otter_nodes_test.py:186]: Same issue here.

File: src/psyclone/tests/psyir/nodes/otter_nodes_test.py [hunk 3]
```diff
@@ -243,11 +208,9 @@ def test_ottersyncorhonisechildrennode_lower_to_language():
...
     calls = routine.walk(Call)
     assert len(calls) == 1
-    assert calls[0].routine.name == "fortran_otterSynchroniseChildTasks_i"
-    assert len(calls[0].children) == 3
-    assert calls[0].children[0].name == "__FILE__"
-    assert calls[0].children[1].value == "my_routine"
-    assert calls[0].children[2].name == "__LINE__"
+    assert calls[0].routine.name == "fortran_otterSynchroniseTasks_i"
+    assert len(calls[0].children) == 1
+    assert calls[0].children[0].value == "0"
```
Comment (LonelyCat124) [inline src/psyclone/tests/psyir/nodes/otter_nodes_test.py:213]: Again here

File: src/psyclone/tests/psyir/nodes/otter_nodes_test.py [hunk 4]
```diff
@@ -272,10 +235,7 @@ def test_ottersyncorhonisedescendenttasksnode_lower_to_language():
     assert len(calls) == 2
     assert (calls[0].routine.name == 
             "fortran_otterSynchroniseDescendantTasksBegin_i")
-    assert len(calls[0].children) == 3
-    assert calls[0].children[0].name == "__FILE__"
-    assert calls[0].children[1].value == "my_routine"
-    assert calls[0].children[2].name == "__LINE__"
+    assert len(calls[0].children) == 0
```
Comment (LonelyCat124) [inline src/psyclone/tests/psyir/nodes/otter_nodes_test.py:238]: and here.

File: src/psyclone/tests/psyir/transformations/otter_transformations_test.py [hunk 5]
```diff
@@ -155,9 +136,9 @@ def test_otterloop_trans_apply():
             "fortran_otterLoopEnd, fortran_otterLoopIterationBegin_i, "
             "fortran_otterLoopIterationEnd" in code)
     correct = \
-        '''CALL fortran_otterLoopBegin_i(__FILE__, 'invoke_0_compute_cu', __LINE__)
```
Comment (LonelyCat124) [inline src/psyclone/tests/psyir/transformations/otter_transformations_test.py:158]: This test also fails, the arguments are still produced in the code.

File: src/psyclone/tests/psyir/transformations/otter_transformations_test.py [hunk 6]
```diff
@@ -210,7 +191,7 @@ def test_ottersyncdec_trans_apply():
             "fortran_otterSynchroniseDescendantTasksBegin_i, "
             "fortran_otterSynchroniseDescendantTasksEnd" in code)
     correct = \
-        '''CALL fortran_otterSynchroniseDescendantTasksBegin_i(__FILE__, 'invoke_0_compute_cu', __LINE__)
```
Comment (LonelyCat124) [inline src/psyclone/tests/psyir/transformations/otter_transformations_test.py:213]: Arguments still produced in the output code but not in the test.

File: src/psyclone/tests/psyir/transformations/otter_transformations_test.py [hunk 7]
```diff
@@ -185,8 +166,8 @@ def test_ottersyncchild_trans_apply():
     synctrans.apply(schedule.children[0])
     code = str(psy.gen)
     correct = '''END DO
-      CALL fortran_otterSynchroniseChildTasks_i(__FILE__, 'invoke_0_compute_cu', __LINE__)'''
```
Comment (LonelyCat124) [inline src/psyclone/tests/psyir/transformations/otter_transformations_test.py:188]: Arguments still produced in the output code but not in the test.

**Review by LonelyCat124 — CHANGES_REQUESTED** [review]
Hi @adamtuft , a few tests are failing in `src/psyclone/tests/psyir/transformations/otter_transformations_test.py` and `src/psyclone/tests/psyir/nodes/otter_nodes_test.py`

The root of the issue appears to be to do with the arguments to the calls. Do these calls no longer expect `__FILE__, 'functioncall', __LINE__` as arguments? If not we'll need to make additional changes in PSyclone - but unless there have been other changes in Otter I'd missed I think they still will for the Fortran version.

**Comment by LonelyCat124** [comment]
With all the changes to Otter I think we're going to try to swap to using PSyData to do the otter profiling. I'm going to leave this PR open for now while we work on that but this will be closed later

**Pull request closed without merging**